vircadia/vircadia
# Model Entities: useOriginalPivot Property

implements #284 resolves https://github.com/vircadia/vircadia/issues/793

adds a new property to Model Entities: useOriginalPivot

defaults to false (current behavior), but adds option to toggle in new model dialog

caveat: in create, the bounding box of a selected entity is centered around the actual entity position, which is now offset from the model's position when useOriginalPivot is true.  I actually like this, as it reflects that offset, but maybe others have different opinions

Test plan:
- run this script: https://gist.github.com/HifiExperiments/7b3faa7919eed7a308c762ffe03d893f/raw/615f87b8315e433ac35b23e47a52ab6954e27f6b/PivotTest.js
- first and third models have useOriginalPivot false, second and fourth are true.  First and second models use natural dimensions, third and fourth use { 1, 1, 2 }
- enable Developer -> Physics -> Show Bullet Collisions.  green collision lines should match the 4 models
- hover over the entities with your mouse.  a small white sphere will appear when your mouse is over the entities.  make sure it lines up with the models
- press "o" to toggle through different registrationPoints for all of the models.  note that they show the correct behavior, and that the physics and ray picking continue to work on all 4 models
- verify the workflow of adding a model with/without "Use Original Pivot" via create, and that you can toggle it on a created model via the property window
- run luci.js and use the Bounding Box section to confirm that the render bounds of each model is in the correct place, encompassing it regardless of pivot
